server/core/src/main/java/org/infinispan/server/core/transport/NettyChannelInitializer.java

```diff
@@ -53,7 +56,9 @@ public void initializeChannel(Channel ch) throws Exception {
 
          pipeline.addLast("sni", new SniHandler(domainMappingBuilder.build()));
       }
-      pipeline.addLast("decoder", server.getDecoder());
+      if(decoder != null) {
+         pipeline.addLast("decoder", server.getDecoder());
```

Comment:
Unfortunately this can not be done :( `server.getDecoder()` creates new instance for each call and it seems the decoder can not be shared between piplines. I created (and assigned to you) [ISPN-7765](https://issues.jboss.org/browse/ISPN-7765).